extension/lib/shopgate/CartExtensionPipeline.js

```diff
@@ -86,6 +88,30 @@ class ShopgateCartExtensionPipeline {
     return this._bigCommerceCartRepository.getCheckoutUrl()
   }
 
+  /**
+   * @param {ShopgateUpdateProduct[]} cartItems
+   * @return {Promise<boolean>}
+   */
+  async updateProducts (cartItems) {
+    let nonBreakingErrorsPresent = false
```

Comment:
It is now that I'm noticing how confusing this relation on the boolean result may be.
You may rename this to warningFree for example and revert true/false assignement.
This will require you to also revert condition in the extension/lib/updateProductsInCart.js line 12. You may also rename the variable there to just "success".

Reply:
I changed the polarity of the updateProducts method. It now returns true if there were no errors and false if there were errors.